testing/test_setuponly.py

```diff
@@ -48,7 +51,7 @@ def test_arg1(arg_session, arg_function):
     '''
     )
 
-    result = testdir.runpytest(mode, p)
+    result = pytester.runpytest(mode, str(p))
```

Comment:
Here and in other places below, the `str()` call shouldn't be necessary. This is because `pytester.runpytest` also accepts [`os.PathLike[str]`](https://docs.python.org/3/library/os.html#os.PathLike), which in particular allows a `pathlib.Path`.